guillotina/json/deserialize_value.py

```diff
@@ -37,18 +40,32 @@ def schema_compatible(value, schema_or_field, context=None):
         raise ValueDeserializationError(schema_or_field, value, "Deserializer not found for field")
 
 
+def _optimized_lookup(value, field, context):
+    if field._type in (int, str, float, bool):
+        # for primitive types, all we really do is return the value back.
+        # this is costly for all the lookups
+        if not isinstance(value, field._type):
+            raise WrongType(value, field._type, field.__name__)
+        return value
+    else:
+        return schema_compatible(value, field, context)
```

Comment:
Question: do you think that it could be possible to cache the result of `get_adapter()` for given a schema or field? So we can have a dict of fields mapped to a callable (the adapter) so we don't need to call the get_adapter (that it seems what's slowing down the system)?

Reply:
In other tests that I've done, the adapter lookups are about as fast as dictionary lookups so there wouldn't be much gain there.

The real gain was actually in the adapter itself--if it was a field that didn't require any additional serialization.

I'll run more tests though. It's possible this `_optimized_lookup` isn't worth it anymore. The biggest gain in the module was removing the `field.from_unicode(value)` call which called `validate`--it was done on every sub field.

Reply:
Ok! :-)